docs/api/api.md

```diff
@@ -1,30 +1,42 @@
 # Full API list
 <!--these are now just the APIs - do I want to add the tutorials?-->
 
+This document lists Mbed OS components supported APIs. In general, it lists both common APIs (RTOS and the bare metal profile supported), and RTOS only supported APIs, but occasionally, It has either one.
+
 ## RTOS and event handling
```

Comment:
1. Evelyne, Bartek and Andy have just confirmed that it's "full profile" and "bare metal profile", not "RTOS profile". I will edit this page to reflect that.
1. I'm going to talk to the UX team about how best to show which APIs are full profile only, and which are also for bare metal. I think the multiple headers will confuse users.